packages/amplify-category-auth/src/provider-utils/awscloudformation/auth-stack-builder/auth-cognito-stack-builder.ts

```diff
@@ -1249,4 +1256,32 @@ export class AmplifyAuthCognitoStack extends cdk.Stack implements AmplifyAuthCog
       });
     });
   };
+
+  createHostedUIDomainResource = (props: CognitoStackOptions) => {
+    if (!props.hostedUIDomainName) {
+      return;
+    }
+
+    const authCfnTemplatePath = getResourceCfnTemplatePath(getBackendDirPath(), 'auth', props.resourceName);
+    const authCfnTemplate: Template | undefined = readJson(authCfnTemplatePath, { throwIfNotExist: false });
+    const lambdaCalloutCreatedInCloud = authCfnTemplate?.Resources?.HostedUICustomResource?.Type === 'AWS::Lambda::Function';
+    const userPoolDomainCreatedInCloud = authCfnTemplate?.Resources?.HostedUIDomainResource?.Type === 'AWS::Cognito::UserPoolDomain';
+
+    if (lambdaCalloutCreatedInCloud && !userPoolDomainCreatedInCloud) {
+      this.deleteExistingHostedUICustomResource();
+    }
+
+    this.hostedUIDomainResource = new cognito.CfnUserPoolDomain(this, 'HostedUIDomainResource', {
+      domain: cdk.Fn.conditionIf(
+        'ShouldNotCreateEnvResources',
+        cdk.Fn.ref('hostedUIDomainName'),
+        cdk.Fn.join('-', [cdk.Fn.ref('hostedUIDomainName'), cdk.Fn.ref('env')]),
+      ).toString(),
+      userPoolId: cdk.Fn.ref('UserPool'),
+    });
+
+    if (this.hostedUICustomResource) {
```

Comment:
Can we just remove this hostedUICustomResource ?

Reply:
Not yet, because the custom resource has to delete the existing resource first

Reply:
if we just remove the custom resource, cfn will send a delete request to lambda and that will flush the resource or (Userpool domain)

Reply:
@akshbhu you're saying if I delete the custom resource, it will delete the existing user pool domain? When I tested this, that wasn't the case. What happened was I deleted the Lambda and the custom resource, the user pool domain stuck around, and I had to remove it manually with the lambda before I could recreate it with CFN.

Reply:
okay